core/emu.py

```diff
@@ -0,0 +1,306 @@
+#!/usr/bin/env python3
+import gzip
+import os
+import platform
+import signal
+import subprocess
+import sys
+import tempfile
+import time
+from pathlib import Path
+
+import click
+
+import trezorlib.debuglink
+import trezorlib.device
+from trezorlib._internal.emulator import CoreEmulator
+
+try:
+    import inotify.adapters
+except ImportError:
+    inotify = None
+
+
+HERE = Path(__file__).parent.resolve()
+MICROPYTHON = HERE / "build" / "unix" / "micropython"
+SRC_DIR = HERE / "src"
+SD_CARD_GZ = HERE / "trezor.sdcard.gz"
+
+PROFILING_WRAPPER = HERE / "prof" / "prof.py"
+
+PROFILE_BASE = Path.home() / ".trezoremu"
+
+
+def run_command_with_emulator(emulator, command):
+    with emulator:
+        # first start the subprocess
+        process = subprocess.Popen(command)
+        # After the subprocess is started, ignore SIGINT in parent
+        # (so that we don't need to handle KeyboardInterrupts)
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
+        # SIGINTs will be sent to all children by the OS, so we should be able to safely
+        # wait for their exit.
+        return process.wait()
+
+
+def run_emulator(emulator):
+    with emulator:
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
+        return emulator.wait()
+
+
+def watch_emulator(emulator):
+    watch = inotify.adapters.InotifyTree(str(SRC_DIR))
+    try:
+        for _, type_names, _, _ in watch.event_gen(yield_nones=False):
+            if "IN_CLOSE_WRITE" in type_names:
+                click.echo("Restarting...")
+                emulator.restart()
+    except KeyboardInterrupt:
+        emulator.stop()
+    return 0
+
+
+def run_debugger(emulator):
+    os.chdir(emulator.workdir)
+    env = emulator.make_env()
+    if platform.system() == "Darwin":
+        env["PATH"] = "/usr/bin"
+        os.execvpe(
+            "lldb",
+            ["lldb", "-f", emulator.executable, "--"] + emulator.make_args(),
+            env,
+        )
+    else:
+        os.execvpe(
+            "gdb", ["gdb", "--args", emulator.executable] + emulator.make_args(), env
+        )
+
+
+@click.command(context_settings=dict(ignore_unknown_options=True))
+# fmt: off
```

Comment:
it's formatted manually. I don't want the Black formatting here because it would be inconsistent between lines that fit like so
```
@click.option(
    "-a", "--ahoj-babi", help="ahoj babi"
)
```
and
```
@click.option(
	"-a",
 	"--ahoj-babi",
)
```

Reply:
I would prefer having everything on the same line then. But your choice

Reply:
formatting changed to one option per line